File: readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

double grader(string s); // function prototype

int main(void)
{
    string text = get_string("Text: \n"); // accepts string from user
    double grade = round(grader(text));   // rounding of grade to nearest whole number

    // grade conditionals
    if (grade < 1)
        printf("Before Grade 1\n");
    else if (grade > 16)
        printf("Grade 16+\n");
    else
        printf("Grade %i\n", (int) grade);
}

double grader(string txt)
{
    float letters = 0, sentences = 0, words = 1; // float because int divided by int gives int

    // counting no of letters, words and sentences
    for (int i = 0, n = strlen(txt); i < n; i++)
    {
        if (isalnum(txt[i]))
            letters++;
        else if ((txt[i] == '.') || (txt[i] == '!') || (txt[i] == '?'))
            sentences++;
        else if (isspace(txt[i]))
            words++;
    }

    // calculating grade
    float l = (letters / words) * 100;
    float s = (sentences / words) * 100;
    double index = 0.0588 * l - 0.296 * s - 15.8;

    return index;
}

File: week6/readability.py
#  input text from user
text = input("Text: ")

letter_count = 0
sent_count = 0
word_count = 1

# traversing through each character in text
for i in text:
    # if i is an alphabet, letter count is incremented
    if i.isalpha():
        letter_count += 1
    # if i is an punctiation mark ending sent, sent count is incremented
    elif i in [".", "?", "!"]:
        sent_count += 1
    # if i is an space, word count is incremented
    elif i.isspace():
        word_count += 1

# calculating average number of letters per 100 words
L = letter_count / (word_count/100)
# calculating average number of sentences per 100 words
S = sent_count / (word_count/100)

# calculating grade using Coleman-Liau index
grade = round(0.0588 * L - 0.296 * S - 15.8)

# grade conditionals
if grade < 1:
    print("Before Grade 1")
elif grade >= 16:
    print("Grade 16+")
else:
    print(f"Grade {grade}")
